### PR TITLE
fix SRVKP-4454: added check to skip over missing key in JSON data

### PR DESCRIPTION
**Pull request by rh-rahulshetty**
This PR resolves the following issue: [SRVKP-4454](https://issues.redhat.com/browse/SRVKP-4454).

### CODE REVIEW

File: tools/compare-TaskRun-Pod-creationTimestamp.py [hunk 1]
```diff
@@ -35,14 +35,17 @@ def doit(args, status_data):
...
+            tr_name = i["metadata"]["name"]
+            tr_creationTimestamp = i["metadata"]["creationTimestamp"]
+            tr_podName = i["status"]["podName"]
+            assert tr_name not in data_taskruns
+            data_taskruns[tr_name] = {
+                "creationTimestamp": tr_creationTimestamp,
+                "podName": tr_podName,
+            }
+        except KeyError as e:
+            logging.debug(f"Missing key details in payload: {e}")
```
Comment (jhutar) [inline tools/compare-TaskRun-Pod-creationTimestamp.py:48]: Maybe make this a warning?

**Review by jhutar — CHANGES_REQUESTED** [review]
Just small comment added. If you do not want to fix it, let me know and we can merge as it is.

File: tools/compare-TaskRun-Pod-creationTimestamp.py (hunk 1 above)
Comment (rh-rahulshetty) [reply tools/compare-TaskRun-Pod-creationTimestamp.py]: Done

File: tools/compare-TaskRun-Pod-creationTimestamp.py [hunk 2]
```diff
@@ -35,14 +35,17 @@ def doit(args, status_data):
...
+        try:
+            tr_name = i["metadata"]["name"]
+            tr_creationTimestamp = i["metadata"]["creationTimestamp"]
+            tr_podName = i["status"]["podName"]
+            assert tr_name not in data_taskruns
+            data_taskruns[tr_name] = {
+                "creationTimestamp": tr_creationTimestamp,
+                "podName": tr_podName,
+            }
+        except KeyError as e:
```
[the remark this answers is not in the document]
Comment (rh-rahulshetty) [reply tools/compare-TaskRun-Pod-creationTimestamp.py:42]: Moved the checks to relevant Key lookup statements.

**Comment by rh-rahulshetty** [comment]
Thanks for the inputs @jhutar. I've made the changes in the latest commit.

**Review by jhutar — APPROVED** [review]
/lgtm

**Pull request merged**